### PR TITLE
doc: move module submission requirements

### PR DESCRIPTION
**Pull request by gdams**
This PR moves the module submission requirements from README.md to
CONTRIBUTING.md

##### Checklist

- [x] `npm test` passes
- [x] documentation is changed or added
- [x] contribution guidelines followed [here](https://github.com/nodejs/citgm/blob/master/CONTRIBUTING.md)
- [x] commit message follows commit guidelines

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,34 +151,6 @@ npm run test
 
 This will run both a linter and a tap based unit test suite.
 
-## Requirements for inclusion in Node.js Citgm runs
```
Comment (gibfahn) [inline README.md]: I think it's worth adding a link here, something like:

>If you want to submit a module to be run in the Node.js CI, see the [requirements](./CONTRIBUTING.md)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -44,4 +44,29 @@ Fixes: https://github.com/nodejs/citgm/issues/313
 ```
 
 ## Submitting a module to CitGM
-[Read the module requirements](README.md#requirements-for-inclusion-in-nodejs-citgm-runs)
+
```
Comment (richardlau) [inline CONTRIBUTING.md]: I think some context is needed here. This is for adding a module to be included in the default citgm-all runs and not, for example, for adding dependencies to CitGM.

Also depending on the module it does not need to be added to lookup.json for CitGM to be able to test it.

File: README.md (hunk 1 above)
Comment (gdams) [inline README.md]: added

File: CONTRIBUTING.md (hunk 2 above)
Comment (gdams) [inline CONTRIBUTING.md]: updated PTAL

**Pull request merged**